src/go/cmd/token_builders.go

```diff
@@ -0,0 +1,264 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package cmd
+
+import (
+	"strings"
+)
+
+func addTab() string {
+	return "     "
+}
+
+func makeToken(defID, navID *string, val string, kind int, list *[]Token) {
```

Comment:
It was not clear from invocations below that this mutates state.  Typical pattern is to mimic the semantics of `append()`,  So something like this.
```go
func appendToken(list []Token, defID, navID *string, val string, kind int) []Token { //...}
```
Probably not worth refactoring at this point but something to keep in mind for the future.

Also, comments explaining how this is used would be useful as there's a mix of seemingly optional params (that should really be at the end) and required(?) params.  Maybe a variadic options pattern would work better here?

Reply:
Added a comment above makeToken for a possible future refactoring to improve the function signature